src/plugins/intel_gna/src/serial/gna_model_serial.cpp

```diff
@@ -149,7 +149,7 @@ header_latest::ModelHeader GNAModelSerial::ReadHeader(std::istream& is) {
     auto size_of_headers_header = sizeof(header_latest::ModelHeader::gnam) +
                                   sizeof(header_latest::ModelHeader::headerSize) +
                                   sizeof(header_latest::ModelHeader::Version);
-    if (stream_len > size_of_headers_header) {
+    if (static_cast<uint32_t>(stream_len) > size_of_headers_header) {
```

Comment:
are you sure that stream_len at this point will be >= 0?

Reply:
yes, in order for it to be <0 the stream starting position would have to be greater than stream length which is impossible